pyscripts/mask.py

```diff
@@ -0,0 +1,195 @@
+    "server_certs_key_passphrase",
+]
+
+CONNECTION_KEYS = ["rabbit", "database_connection",
+                   "slave_connection", "sql_connection"]
+# Masking string
+MASK_STR = "**********"
+
+# general and connection regexes are used to match the pattern that should be
+# applied to both Protect keys and connection keys, which is the same thing
+# done in SoS reports
+gen_regex = r'(\w*(%s)\s*=\s*)(.*)' % "|".join(PROTECT_KEYS)
+con_regex = r'((%s)\s*://)(\w*):(.*)(@(.*))' % "|".join(CONNECTION_KEYS)
+
+# In k8s secrets, it's possible to define sensitive information in the form
+# of <key> = <value> under the "Data" section. This kind of info is not matched
+# within a config file, hence we need to parse keys and detect any potential
+# data that matches the keys we want to protect/mask.
+key_regex = r'(%s)$' % "|".join(PROTECT_KEYS)
+
+regexes = [gen_regex, con_regex]
+
+
+class SecretMask():
+    """
+    Given a path to k8s secret containing sensitive base64 encoded data,
+    this class implements the required functions to load, unpack the secret
+    content, mask any potential sensitive data and pack it again.
+    The resulting file won't expose any information that should be protected
+    by the defined pattern.
+    """
+
+    def __init__(self, path: Optional[str] = None) -> None:
+        self.path: Union[str, None] = path
+
+    def mask(self) -> bool:
+        """
+        Read a k8s secret dumped as yaml and process the
+        Data section: for each entry analyze the resulting
+        dict and mask any sensitive info if the pattern is
+        matched.
+        """
+        s = self._readYaml()
+        # s is None or empty dict, return
+        if not s or len(s) == 0:
+            return True
+        for k, v in s.items():
+            # if we have items in the loaded dict,
+            # we look for the data section, which
+            # is were we want to apply masking
+            if k == "data":
+                data = self._process_data(v)
+                s[k] = data
+        # write the resulting, masked/encoded file
+        self._writeYaml(dict(s))
+        return True
+
+    def _readYaml(self) -> Dict[str, str]:
+        """
+        Read and Load the k8s Secret dumped as
+        yaml file.
+        """
+        s = dict()
+        try:
+            assert self.path is not None
+            with open(self.path, 'r') as f:
+                s = yaml.safe_load(f)
+            return s
+        except FileNotFoundError:
+            print("File doesn not exist")
+            sys.exit(-1)
+        return s
+
+    def _writeYaml(self, encoded_secret: Any) -> None:
+        """
+        Re-write the masked secret in the same
+        path.
+        """
+        try:
+            assert self.path is not None
+            with open(self.path, 'w') as f:
+                yaml.safe_dump(encoded_secret, f,
+                               default_flow_style=False)
+        except IOError:
```

Comment:
Similiar comment here:

~~~
    except (IOError, yaml.YAMLError) as e:
        print(f"Error while writing the masked file: {e}")
~~~

Reply:
Done